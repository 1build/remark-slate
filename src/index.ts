import plugin, { transform, defaultNodeTypes } from './serializer';

export { transform, defaultNodeTypes };

export default plugin;
