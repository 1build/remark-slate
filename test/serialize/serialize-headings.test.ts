import { serialize, defaultNodeTypes } from '../../src';

it('Serialize a heading one from slate state to markdown', () => {
  expect(
    serialize({
      type: defaultNodeTypes.heading[1],
      children: [
        {
          text: 'Heading one',
        },
      ],
    })
  ).toMatchSnapshot();
});

it('Serialize a heading two from slate state to markdown', () => {
  expect(
    serialize({
      type: defaultNodeTypes.heading[2],
      children: [
        {
          text: 'Heading two',
        },
      ],
    })
  ).toMatchSnapshot();
});

it('Serialize a heading three from slate state to markdown', () => {
  expect(
    serialize({
      type: defaultNodeTypes.heading[3],
      children: [
        {
          text: 'Heading three',
        },
      ],
    })
  ).toMatchSnapshot();
});

it('Serialize a heading four from slate state to markdown', () => {
  expect(
    serialize({
      type: defaultNodeTypes.heading[4],
      children: [
        {
          text: 'Heading four',
        },
      ],
    })
  ).toMatchSnapshot();
});

it('Serialize a heading five from slate state to markdown', () => {
  expect(
    serialize({
      type: defaultNodeTypes.heading[5],
      children: [
        {
          text: 'Heading five',
        },
      ],
    })
  ).toMatchSnapshot();
});

it('Serialize a heading six from slate state to markdown', () => {
  expect(
    serialize({
      type: defaultNodeTypes.heading[6],
      children: [
        {
          text: 'Heading six',
        },
      ],
    })
  ).toMatchSnapshot();
});
