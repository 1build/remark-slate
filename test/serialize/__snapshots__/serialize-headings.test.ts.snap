// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serialize a heading five from slate state to markdown 1`] = `
"##### Heading five
"
`;

exports[`Serialize a heading four from slate state to markdown 1`] = `
"#### Heading four
"
`;

exports[`Serialize a heading one from slate state to markdown 1`] = `
"# Heading one
"
`;

exports[`Serialize a heading six from slate state to markdown 1`] = `
"###### Heading six
"
`;

exports[`Serialize a heading three from slate state to markdown 1`] = `
"### Heading three
"
`;

exports[`Serialize a heading two from slate state to markdown 1`] = `
"## Heading two
"
`;
